rules/hmap.bzl

```diff
@@ -50,33 +20,31 @@ def _make_headermap_impl(ctx):
     :return: provider with the info for this rule
     """
 
-    # Write a file for *this* headermap, this is a temporary file
-    input_f = ctx.actions.declare_file(ctx.label.name + "_input.txt")
-    all_hdrs = list(ctx.files.hdrs)
+    # Add a list of headermaps in text or hmap format
+    args = ctx.actions.args()
+    if ctx.attr.namespace:
+        args.add("--namespace", ctx.attr.namespace)
+
+    args.add("--output", ctx.outputs.headermap.path)
+
+    args.add_all(ctx.files.hdrs)
     for provider in ctx.attr.direct_hdr_providers:
         if apple_common.Objc in provider:
-            all_hdrs += provider[apple_common.Objc].direct_headers
+            args.add_all(provider[apple_common.Objc].direct_headers)
         elif CcInfo in provider:
-            all_hdrs += provider[CcInfo].compilation_context.direct_headers
+            args.add_all(provider[CcInfo].compilation_context.direct_headers)
         else:
             fail("direct_hdr_provider %s must contain either 'CcInfo' or 'objc' provider" % provider)
 
-    out = _make_headermap_input_file(ctx.attr.namespace, all_hdrs, ctx.attr.flatten_headers)
-    ctx.actions.write(
-        content = out,
-        output = input_f,
-    )
-
-    # Add a list of headermaps in text or hmap format
-    inputs = [input_f]
-    args = [input_f.path, ctx.outputs.headermap.path]
+    args.set_param_file_format(format = "multiline")
+    args.use_param_file("@%s", use_always = True)
```

Comment:
I don't think we need `use_always` anymore?

Reply:
```suggestion
    args.use_param_file("@%s")
```
Yeah